src/dotty/tools/dotc/sbt/ThunkHolder.scala

```diff
@@ -0,0 +1,59 @@
+package dotty.tools.dotc
+package sbt
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+import xsbti.api
+
+/** Create and hold thunks */
+private[sbt] trait ThunkHolder {
+  private[this] val thunks = new ListBuffer[api.Lazy[_]]
+
+  /** Force all unevaluated thunks to prevent space leaks. */
+  @tailrec protected final def forceThunks(): Unit = if (!thunks.isEmpty) {
+    val toForce = thunks.toList
+    thunks.clear()
+    toForce.foreach(_.get())
+    // Forcing thunks may create new thunks
+    forceThunks()
+  }
+
+  /** Store the by-name parameter `s` in a `Lazy` container without evaluating it.
+   *  It will be forced by the next call to `forceThunks()`
+   */
+  def lzy[T <: AnyRef](t: => T): api.Lazy[T] = {
+    val l = SafeLazy(() => t)
+    thunks += l
+    l
+  }
+
+  /** Store the parameter `s` in a `Lazy` container, since `s` is not by-name, there
+   *  is nothing to force.
+   *
+   *  TODO: Get rid of this method. It is only needed because some xsbti.api classes
+   *  take lazy arguments when they could be strict, but this can be fixed in sbt,
+   *  see https://github.com/sbt/zinc/issues/114
+   */
+  def strictlzy[T <: AnyRef](t: T): api.Lazy[T] =
```

Comment:
> I'm confused by your suggestion, I can't have an overloaded method where the only difference is that the argument is T or => T, they would conflict with each other.

Indeed. I was assuming that we can follow the code pattern in [Ensuring](https://github.com/scala/scala/blob/2.12.x/src/library/scala/Predef.scala#L315-L317). But without an argument this would indeed lead to ambiguous overload. 

>strictLzy

`strict2lzy` would be even easier to understand.